### DISCUSSION TITLE
setSession doesn't call _notifyAllSubscribers, is it correct?

### DISCUSSION THREAD

**Discussion by martinboksa** [supabase/supabase#10833] [Questions] [answered]
Hello guys,
I made an electron app where users can sign in with otp

```typescript
await supabase.auth.signInWithOtp({
      email,
      options: { shouldCreateUser: true, emailRedirectTo },
    })
```

When the user clicks on `magic link`, it redirects to the web page and opens the electron with `access_token` and `refresh_token`

Then electron tries to `setSession` 

```typescript
 supabase.auth
        .setSession({
          access_token: accessToken,
          refresh_token: refreshToken,
        })
```

**The app is subscribed to `supabase.auth.onAuthStateChange` but `setSession` doesn't call `_notifyAllSubscribers`, so app cannot recognize when the user is signed in.**

**For now I made workaround**
```typescript
      supabase.auth
        .setSession({
          access_token: accessToken,
          refresh_token: refreshToken,
        })
        .then(({ data: { session } }) => {
          if (session) {
            // @ts-ignore set session does not call subscribers when session is updated
            supabase.auth._notifyAllSubscribers('SIGNED_IN', session)
          }
        })
```

So question is:
1. Why `setSession` doesnt fire `_notifyAllSubscribers`? 
2. Any idea for better workaround, please?

Thank you guys!

**Comment by ahofmeister** [comment]
Thank you so much for the workaround. I don't like it either but at least it works. Cost me a couple of days :D

**Comment by j4w8n** [comment]
@martinboksa @ahofmeister There is now a PR to trigger that event. We'll see if it gets merged.
https://github.com/supabase/gotrue-js/pull/581

**Comment by j4w8n** [comment]
@martinboksa @ahofmeister can you help me understand why setSession is required with Electron, instead of the user automatically being logged in after clicking the otp link?

**Reply by j4w8n** [reply]
Disregard my last ask. PR was merged. gotrue-js version will likely be bumped to 2.8.0 in the next release of supabase-js next week.

**Comment by hf** [comment, marked as answer]
Will be available in the next version of gotrue-js.
